pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -0,0 +1,316 @@
+			}
+			return fmt.Errorf("failed to get imageStreamTag %s from registry cluster: %w", req.NamespacedName.String(), err)
+		}
+		isTags[ctx] = imageStreamTag
+	}
+
+	srcContext, err := findNewest(isTags)
+	if err != nil {
+		// if it doesn't exist on both clusters; only log the error
+		log.Warnf("failed to find the new isTag from %v: %v", isTags, err)
+		return nil
+	}
+	sourceImageStreamTag := isTags[srcContext]
+
+	imageStreamNameAndTag := strings.Split(req.Name, ":")
+	if n := len(imageStreamNameAndTag); n != 2 {
+		return fmt.Errorf("when splitting imagestreamtagname %s by : expected two results, got %d", req.Name, n)
+	}
+	imageStreamName, imageTag := imageStreamNameAndTag[0], imageStreamNameAndTag[1]
+	sourceImageStream := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageStreamName,
+			Namespace: req.Namespace,
+		},
+	}
+
+	*log = *log.WithField("docker_image_reference", sourceImageStreamTag.Image.DockerImageReference)
+
+	for ctx, isTag := range isTags {
+		if ctx == srcContext {
+			continue
+		}
+		client := r.registryClients[ctx]
+		if err := client.Get(r.ctx, types.NamespacedName{Name: req.Namespace}, &corev1.Namespace{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to check if namespace %s exists: %w", req.Namespace, err)
+			}
+			if err := client.Create(r.ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: req.Namespace}}); err != nil && !apierrors.IsAlreadyExists(err) {
+				return fmt.Errorf("failed to create namespace %s: %w", req.Namespace, err)
+			}
+		}
+
+		if err := r.ensureImageStream(sourceImageStream, client, log); err != nil {
+			return fmt.Errorf("failed to ensure imagestream: %w", err)
+		}
+
+		if isTag != nil && isTag.Image.Name == sourceImageStreamTag.Image.Name {
+			log.Debug("ImageStreamTag is current")
+			return nil
+		}
+
+		if err := r.ensureImagePullSecret(req.Namespace, client, log); err != nil {
+			return fmt.Errorf("failed to ensure imagePullSecret: %w", err)
+		}
+		imageName, err := publicDomainForImage(srcContext, sourceImageStreamTag.Image.DockerImageReference)
+		if err != nil {
+			return fmt.Errorf("failed to get the public domain: %w", err)
+		}
+
+		imageStreamImport := &imagev1.ImageStreamImport{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: req.Namespace,
+				Name:      imageStreamName,
+			},
+			Spec: imagev1.ImageStreamImportSpec{
+				Import: true,
+				Images: []imagev1.ImageImportSpec{{
+					From: corev1.ObjectReference{
+						Kind: "DockerImage",
+						Name: imageName,
+					},
+					To: &corev1.LocalObjectReference{Name: imageTag},
+					ReferencePolicy: imagev1.TagReferencePolicy{
+						Type: imagev1.LocalTagReferencePolicy,
+					},
+				}},
+			},
+		}
+
+		r.log.Debugf("creating import %v on cluster %s", imageStreamImport, ctx)
+
+		// ImageStreamImport is not an ordinary api but a virtual one that does the import synchronously
+		if err := client.Create(r.ctx, imageStreamImport); err != nil {
```

Comment:
It might make sense to factor this until the status check out, because we want a success or error counter metric

Reply:
Done.